fem/fespace.cpp

```diff
@@ -3866,6 +4031,9 @@ void FiniteElementSpace::Save(std::ostream &os) const
    }
 }
 
+std::shared_ptr<const PRefinementTransferOperator>
+FiniteElementSpace::GetPrefUpdateOperator() { return std::move(PTh); }
```

Comment:
My understanding is that this will transfer the ownership of the underlying pointer to `shared_ptr`, which will subsequently be deleted after it is used in `GridFunction::UpdatePRef()`. If the user intends to update multiple grid-functions, the second call to `UpdatePRef()` will error out.

Reply:
Good catch. I changed `PTh` to a `shared_ptr`, to avoid this issue. Testing in `ex6p` confirmed that having a second `ParGridFunction` with `UpdatePRef()` was failing, and now it works. However, this points out that the interface confusingly has `UpdatePRef` and simply `Update` (see hpref.cpp), with only the former constructing `PTh`. I should document this at least, or have `Update` also construct `PTh`. Another awkward part of the interface I realize now is that we have `VarOrderElemInfo` as input to `UpdatePRef` and `pRefinement` as input to `ChangeElementOrders`. This should be simplified.

Reply:
I committed a simplification. There are many choices for the interface, but I chose to remove the redundant `ChangeElementOrders` function and use `pRefinement` rather than `VarOrderElemInfo` for `UpdatePRef`. Also, `UpdatePRef` has an argument now to specify whether to maintain the old space and the transfer operator.